handlers/geolocation/geolocation.go

```diff
@@ -200,6 +203,10 @@ func (c *GeolocationHandlersCollection) getStreamPull(playbackID string) (string
 		return "", nil
 	}
 
+	if err := c.Lapi.LockPull(stream.ID, lockPullLeaseTimeout); err != nil {
+		return "", fmt.Errorf("failed to lock pull, err=%v", err)
```

Comment:
Do you know if this will result in a playback error going back to the client or is there a retry somewhere?

Reply:
Ah no it will end up defaulting to returning `"push://"` so I think we probably need to either return the error in this case to allow a retry to happen (and check the playback behaviour for that)

Reply:
Updated with the retries. PTAL.